backend/session/direct-ipc.c

```diff
@@ -193,8 +208,11 @@ int direct_ipc_open(int sock, const char *path) {
 
 	send_msg(sock, -1, &msg, sizeof(msg));
 
-	int fd, err;
-	recv_msg(sock, &fd, &err, sizeof(err));
+	int fd, err, ret;
+	int retry = 0;
+	do {
+		ret = recv_msg(sock, &fd, &err, sizeof(err));
+	} while (ret == 0 && retry++ < 3);
```

Comment:
Do you think we could check for `errno` here? (Is it set to `EINTR` or something like that?)